### PR TITLE
[16.0][IMP] stock_inventory: various fixes

### PR DESCRIPTION
**Pull request by DavidJForgeFlow**
Forward port of https://github.com/OCA/stock-logistics-warehouse/pull/1991 https://github.com/OCA/stock-logistics-warehouse/pull/2010. 

Supersedes #2017 

@ForgeFlow

### CODE REVIEW

**Review by LoisRForgeFlow — APPROVED** [review]
Ty

**Review by JoanSForgeFlow — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
I'm merging to align with v15.

/ocabot merge major

File: stock_inventory/views/stock_inventory.xml [hunk 1]
```diff
@@ -150,5 +161,7 @@
         sequence="30"
         action="action_view_inventory_group_form"
     />
-    <delete model="ir.ui.menu" id="stock.menu_action_inventory_tree" />
+    <record id="stock.menu_action_inventory_tree" model="ir.ui.menu">
+        <field name="active" eval="False" />
+    </record>
```
Comment (Kev-Roche) [inline stock_inventory/views/stock_inventory.xml:166]: If we already have the module installed, the view does not exist anymore, so we can not access to "stock.menu_action_inventory_tree" and an error occurs. 
@DavidJForgeFlow @LoisRForgeFlow

Comment (DavidJForgeFlow) [reply stock_inventory/views/stock_inventory.xml:166]: Hi! You can just update the module "stock" and then install / update the "stock_inventory" module!

Comment (Kev-Roche) [reply stock_inventory/views/stock_inventory.xml:166]: Of course, but depending of the odoo hosting infrastructure, it is not a suitable / clean solution. Returning to the previous line of code or a small migration script will work without effort. what do you think ?

Comment (FreezeS) [reply stock_inventory/views/stock_inventory.xml:166]: "You can just update the module "stock" and then install / update the "stock_inventory" module!" - I confirm this works but it's not intuitive or documented anywhere except in this thread. A better solution needs to be found. Is it possible to force the upgrade of "stock" before upgrading this module?

**Pull request merged**